main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"os"
+
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-steplib/steps-restore-gradle-cache/step"
+)
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+	logger := log.NewLogger()
+	envRepo := env.NewRepository()
+	inputParser := stepconf.NewInputParser(envRepo)
+	cacheStep := step.New(logger, inputParser, envRepo)
+
+	exitCode := 0
+
+	if err := cacheStep.Run(); err != nil {
+		logger.Errorf(err.Error())
+		exitCode = 1
+		return exitCode
```

Comment:
Small nit pick: Since the very next execution is a `return`, we could remove this.